ziti/ziti.go

```diff
@@ -486,8 +486,13 @@ func (context *ContextImpl) Sessions() ([]*rest_model.SessionDetail, error) {
 
 func (context *ContextImpl) OnClose(routerConn edge.RouterConn) {
 	logrus.Debugf("connection to router [%s] was closed", routerConn.Key())
-	context.Emit(EventRouterDisconnected, routerConn.GetRouterName(), routerConn.Key())
-	context.routerConnections.Remove(routerConn.Key())
+	removed := context.routerConnections.RemoveCb(routerConn.Key(), func(key string, v edge.RouterConn, exists bool) bool {
+		return exists && v == routerConn
+	})
+
+	if removed {
+		context.Emit(EventRouterDisconnected, routerConn.GetRouterName(), routerConn.Key())
+	}
```

Comment:
Removing this is ...ok?

Reply:
If we create two connections to the router at the same time, and one is a duplicate, we close the duplicate. Should we emit a router disconnect if we close a duplicate connection, or only if we close the connection that's actually being used?

Reply:
I don't have a clear answer. So here are some thoughts I have:

In a perfect world, we would go through some connection coordination pool/requester/etc. to ERs that would handle that for us, so we don't have a race condition. I don't recall enough about how these systems interact to know if that's even a reasonable goal to pursue.

Barring that, if I am in a world with multiple outgoing connections, if we have an existing connection that is serviceable and we don't close/lose it and continue to use it, then not emitting a connect/close is fine. I'm not sure if any tests would react to that - my gut says no.

Reply:
Ah, I missed the connect event. I moved that so that it only gets emitted if we use the new connection. Agree that we should ideally only make one connection at a time, but I don't want to get distracted by refactoring that at the moment.

Reply:
seems reasonable to me? :+1: